src/components/AttachmentCarousel/index.js

```diff
@@ -0,0 +1,260 @@
+        };
+    }
+
+    componentDidMount() {
+        this.makeStateWithReports();
+    }
+
+    componentDidUpdate(prevProps) {
+        const previousReportActionsCount = _.size(prevProps.reportActions);
+        const currentReportActionsCount = _.size(this.props.reportActions);
+        if (previousReportActionsCount === currentReportActionsCount) {
+            return;
+        }
+        this.makeStateWithReports();
+    }
+
+    /**
+     * Helps to navigate between next/previous attachments
+     * @param {Object} attachmentItem
+     * @returns {Object}
+     */
+    getAttachment(attachmentItem) {
+        const source = _.get(attachmentItem, 'source', '');
+        const file = _.get(attachmentItem, 'file', {name: ''});
+        this.props.onNavigate({source: addEncryptedAuthTokenToURL(source), file});
+
+        return {
+            source,
+            file,
+        };
+    }
+
+    /**
+     * Toggles the visibility of the arrows
+     * @param {Boolean} shouldShowArrow
+     */
+    toggleArrowsVisibility(shouldShowArrow) {
+        this.setState({shouldShowArrow});
+    }
+
+    /**
+     * This is called when there are new reports to set the state
+     */
+    makeStateWithReports() {
+        let page;
+        const actions = ReportActionsUtils.getSortedReportActions(_.values(this.props.reportActions), true);
+
+        // This is used to match the initial image URL from props in a config environment.
+        // Eg: while using Ngrok the image path is from an Ngrok URL and not an Expensify URL.
+        const propsSource = this.props.source.replace(
+            Config.EXPENSIFY.EXPENSIFY_URL,
+            Config.EXPENSIFY.URL_API_ROOT,
+        );
+
+        /**
+         * Looping to filter out attachments and retrieve the src URL and name of attachments.
+         */
+        const attachments = [];
+        _.forEach(actions, ({originalMessage, message}) => {
+            // Check for attachment which hasn't been deleted
+            if (!originalMessage || !originalMessage.html || _.some(message, m => m.isEdited)) {
+                return;
+            }
+            const matches = [...originalMessage.html.matchAll(CONST.REGEX.ATTACHMENT_DATA)];
+
+            // matchAll captured both source url and name of the attachment
+            if (matches.length === 2) {
+                const [originalSource, name] = _.map(matches, m => m[2]);
+
+                // Update the image URL so the images can be accessed depending on the config environment.
+                // Eg: while using Ngrok the image path is from an Ngrok URL and not an Expensify URL.
+                const source = originalSource.replace(
+                    Config.EXPENSIFY.EXPENSIFY_URL,
+                    Config.EXPENSIFY.URL_API_ROOT,
+                );
+
+                // Determine attachment index by matching source url with one that's used in state or props
+                if ((this.state.source && source.includes(this.state.source))
```

Comment:
Santhosh brought this to my attention i.e., in some cases the image would not load on desktop. I think the problem is because of this.

This is the data of all attachments in my test,
```
[
    {
        "source": "https://staging.expensify.com/chat-attachments/6969758870019010001/w_7097a98a44c04f55a9f2e07c37eea8293ed92f5c.jpg",
        "file": {
            "name": "37323B16-A7B3-4842-A80F-E5F5AAAF8207.jpeg"
        }
    },
    {
        "source": "https://staging.expensify.com/chat-attachments/9099966137859080689/w_ebd852c036040b3aff95ce7d3412274712a4f83d.jpg",
        "file": {
            "name": "image.jpg"
        }
    },
    {
        "source": "https://www.expensify.com/chat-attachments/2678413048004799581/w_e359c12a07dbf73543fb63045319d1b181c370c4.jpg",
        "file": {
            "name": "IMG_0004.jpeg"
        }
    },
    {
        "source": "https://www.expensify.com/chat-attachments/985172500/w_552ca100150ff12df55f25130ee836a6381bda1f.pdf",
        "file": {
            "name": "Screenshot2022-03-02at12.16.41AM.pdf"
        }
    },
]
```

`this.state.source` - is undefined
`propSource` - is "https://www.expensify.com/chat-attachments/9099966137859080689/w_ebd852c036040b3aff95ce7d3412274712a4f83d.jpg" which is the same as this.props.source. So the replace few lines above, at line 105 doesn't do much for this scenario.
`source` - is each attachments URL

So because  `propSource` uses `www.expensify.com` but source uses `staging.expensify.com` the check `source.includes(propsSource)` fails and hence page is not set. If page is not set getAttachments few lines down will fail. We'll need to fix the logic here.

Reply:
I did a bit more digging and it looks like either staging or production URL should work. The authentication info you add might work for only the domain the user is logged into. Either way it looks like we would have to just update the condition here because once page is assigned a value the rest seem to work fine.

Reply:
I did some testing using staging, dev and ngrok. I think a good replacement of this line would be with,
```jsx
const sourcePath = new URL(source).pathname;
const currentPath = new URL(this.state.source || propsSource).pathname;
if (sourcePath === currentPath) {
    page = attachments.length;
}
```
It seems to work better for the issue santhosh was facing and feels a bit more readable. Thoughts?

Reply:
@chiragsalian I used this snippet but I also had to account for when `URL_API_ROOT` because URL doesn't play well when there's no domain! Also I cleaned up some of the code a bit too. I set the propsSource in the constructor so it's using the state. This is good for loading more images too and having to find a new page number.

Reply:
Thanks for the Investigation @chiragsalian and fixing @JediWattson , it's working now.